File: src/main/java/de/byjoker/myjfql/security/encryptor/Argon2Encryptor.java
package de.byjoker.myjfql.security.encryptor;

import de.mkammerer.argon2.Argon2Advanced;
import de.mkammerer.argon2.Argon2Factory;

import java.nio.charset.StandardCharsets;

public class Argon2Encryptor implements Encryptor {

    final Argon2Advanced argon2 = Argon2Factory.createAdvanced(Argon2Factory.Argon2Types.ARGON2id);

    @Override
    public String encrypt(String s) {
        return new String(argon2.rawHash(10, 1576, 4, "argon2".toCharArray(), s.getBytes(StandardCharsets.UTF_8)));
    }

    @Override
    public String name() {
        return "ARGON2";
    }

}

File: src/main/java/de/byjoker/myjfql/util/Argon2Encryptor.kt
package de.byjoker.myjfql.util

import de.mkammerer.argon2.Argon2Factory
import java.nio.charset.StandardCharsets

class Argon2Encryptor(private val salt: String) : Encryptor {


    private val argon2 = Argon2Factory.createAdvanced(Argon2Factory.Argon2Types.ARGON2id)
    override val name: String = "ARGON2"

    override fun encrypt(s: String) =
        String(argon2.rawHash(10, 1576, 4, salt.toCharArray(), s.toByteArray(StandardCharsets.UTF_8)))

}
